### PR TITLE
fix+test(Geography): quality audit - hash/equals, projections, tile math, coverage

### PR DESCRIPTION
**Pull request by warny**
## Summary
Full quality audit of `Utils.Geography` (AGENTS.md compliance, bug hunting, performance, test coverage). Details in `Utils.Geography/TODO.md`.

**Bugs fixed:**
- `GeoPoint<T>`/`GeoVector<T>`: `GetHashCode` used raw values while `Equals` is tolerance-based (5 decimals) - violated the hash/equals contract, breaking `Dictionary`/`HashSet` usage.
- `GeoPointList<T>`/`GeoPointList2<T>.BoundingBox`: magic-number seed (min=10000, max=0) produced wrong results when all points have negative lat/lon.
- `EquirectangularProjection<T>.MapPointToGeoPoint`: divided by RightAngle/StraightAngle, which isn't the inverse of the forward transform - broke round-trip.
- `MollweidProjection<T>`: Newton solve mixed degrees and radians (wrong derivative by a factor of 180/pi) and diverged at high latitudes due to a poor initial guess. Rewritten in radians with theta0=latitude. **Found only via the new round-trip test.**
- `MapPoint<T>.TileX`/`TileY`: truncated modulo returned negative offsets for negative projected coordinates; fixed with floored modulo (and the `Tile` grid-index property, which had the same truncated-division issue).
- `GeoVector<T>(string, CultureInfo[])`: parsed the input string twice.
- Several nullable-contract violations (`Recenter` returning `null` on a non-nullable type, unguarded `TextInfo` casts, etc.).

**Documented, not fixed:** an architectural gap between `MapPoint`/`Tile`/`RepresentationConverter`'s pixel-scaling formulas (inconsistent with each other, and with the fact that none of the 7 projections emit `[0,1]`-normalized coordinates). No test or caller defines the intended semantics, so this needs a design decision rather than a speculative fix - see `Utils.Geography/TODO.md`.

**Tests added** for classes that had zero coverage: `BoundingBox`, `GeoPointList`/`GeoPointList2`, `MapPosition`, `ProjectedPoint`, `Tile`, `MapPoint`, `CoordinatesUtil`, `RepresentationConverter`, and round-trip tests for all 7 registered projections (Mercator, Equirectangular, Gall-Peters, Miller, Mollweide, Lambert, Stereographic).

## Test plan
- [x] `dotnet test UtilsTest/UtilsTest.Unit.csproj --filter FullyQualifiedName~Geography` - 77 tests pass
- [x] Full `dotnet test UtilsTest/UtilsTest.Unit.csproj` - 3350 pass, 3 ignored (same as before), 0 failures
- [x] `dotnet build Utils.Geography/Utils.Geography.csproj` - no warnings besides one pre-existing, documented CA2260 analyzer note (CRTP design limitation, out of scope)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Utils.Geography/Geography/Model/GeoPoint.cs [hunk 1]
```diff
@@ -331,10 +331,14 @@ public bool Equals(GeoPoint<T>? other)
...
-        /// <inheritdoc/>
+        /// <summary>
+        /// Returns a hash code consistent with the tolerance-based <see cref="Equals(GeoPoint{T})"/>.
+        /// Values are rounded to the comparer's precision (5 decimal places) so that points considered
+        /// equal by <see cref="Equals(GeoPoint{T})"/> after rounding also produce the same hash code.
+        /// </summary>
         public override int GetHashCode()
         {
-            return ObjectUtils.ComputeHash(Latitude, Longitude);
+            return ObjectUtils.ComputeHash(T.Round(Latitude, 5), T.Round(Longitude, 5));
```
[the remark this answers is not in the document]
Comment (warny) [reply Utils.Geography/Geography/Model/GeoPoint.cs]: Confirmed, good catch. This is a real, unavoidable limitation: because FloatingPointComparer<T> is intentionally non-transitive (documented on the class itself), no rounding-based hash can be exactly consistent with a tolerant Equals for every pair of values - values straddling a rounding boundary (like your 1.000004/1.000006 example) will always exist for *some* rounding precision, no matter which precision is chosen. A fully airtight fix would require either a constant hash (correct but destroys lookup performance) or making equality exact instead of tolerant, which would be a larger behavioral change than this audit set out to make.

I've reworded the GetHashCode XML docs on both GeoPoint and GeoVector to state this honestly (practical improvement, not a complete fix) rather than implying full consistency - see 4a9e6728.

Also noted: the pole special-case in Equals ignoring longitude while GetHashCode includes it is consistent with how Equals is defined (any longitude at the pole is the same point), so that asymmetry is intentional, not a bug.

File: Utils.Geography/Geography/Model/GeoVector.cs [hunk 2]
```diff
@@ -469,9 +486,13 @@ public override bool Equals(object? obj)
...
 
-    /// <inheritdoc />
+    /// <summary>
+    /// Returns a hash code consistent with the tolerance-based <see cref="Equals(GeoVector{T})"/>.
+    /// Values are rounded to the comparer's precision (5 decimal places) so that vectors considered
+    /// equal by <see cref="Equals(GeoVector{T})"/> after rounding also produce the same hash code.
+    /// </summary>
     public override int GetHashCode()
-        => ObjectUtils.ComputeHash(Latitude, Longitude, Bearing);
+        => ObjectUtils.ComputeHash(T.Round(Latitude, 5), T.Round(Longitude, 5), T.Round(Bearing, 5));
```
[the remark this answers is not in the document]
Comment (warny) [reply Utils.Geography/Geography/Model/GeoVector.cs]: Same root cause as the GeoPoint comment above: since GeoVector.Equals delegates to the tolerant GeoPoint.Equals plus a tolerant Bearing comparison, the same boundary-straddle scenario applies to Bearing (and transitively to Latitude/Longitude through the base call). A rounding-based hash can reduce these mismatches but can't eliminate them entirely for a non-transitive tolerant equality - see my reply on the GeoPoint comment for the full reasoning.

Updated the GetHashCode XML doc on GeoVector to say this explicitly rather than implying full consistency - see 4a9e6728.

File: Utils.Geography/Geography/Model/GeoPoint.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (warny) [reply Utils.Geography/Geography/Model/GeoPoint.cs]: Follow-up: per the repo owner's direction, this is now fully resolved rather than just documented. Equals rounds latitude/longitude to EqualityPrecision (5 decimal places) *before* comparing, using the exact same rounding GetHashCode already applies - so both members are always consistent by construction. The accepted trade-off (explicitly requested) is that two values extremely close to each other but on opposite sides of a rounding boundary (e.g. 1.0000449999 vs 1.0000450001) are now treated as unequal, rather than "equal within a tolerance window" as before. See 47fe4348, with new regression tests covering both the equal-hash case and the not-equal-across-a-boundary case.

File: Utils.Geography/Geography/Model/GeoVector.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (warny) [reply Utils.Geography/Geography/Model/GeoVector.cs]: Same fix applied here: GeoVector.Equals now rounds Bearing to EqualityPrecision before comparing (exact match), so it stays consistent with the rounded GetHashCode by construction. See 47fe4348.
